### ISSUE TITLE
NPM install fails with Node 4.0

### ISSUE THREAD

**Issue by hussion** [yfinkelstein/node-zookeeper#114]
make: Entering directory `/data/dev/zk/node_modules/zookeeper/build'
  ACTION binding_gyp_libzk_target_build_zk_client_lib .
Downloading zookeeper-3.4.3 from http://apache.mirrors.tds.net/zookeeper/zookeeper-3.4.3/zookeeper-3.4.3.tar.gz

gzip: stdin: not in gzip format
tar: Child returned status 1
tar: Error is not recoverable: exiting now
Unable to build zookeeper library
make: *** [.] Error 1
make: Leaving directory `/data/dev/zk/node_modules/zookeeper/build'
gyp ERR! build error 
gyp ERR! stack Error: `make` failed with exit code: 2
gyp ERR! stack     at ChildProcess.onExit (/root/.nvm/versions/node/v4.0.0/lib/node_modules/npm/node_modules/node-gyp/lib/build.js:270:23)
gyp ERR! stack     at emitTwo (events.js:87:13)
gyp ERR! stack     at ChildProcess.emit (events.js:172:7)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:200:12)
gyp ERR! System Linux 2.6.32-504.30.3.el6.x86_64
gyp ERR! command "/root/.nvm/versions/node/v4.0.0/bin/node" "/root/.nvm/versions/node/v4.0.0/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "rebuild"
gyp ERR! cwd /data/dev/zk/node_modules/zookeeper
gyp ERR! node -v v4.0.0
gyp ERR! node-gyp -v v3.0.1
gyp ERR! not ok 
npm ERR! Linux 2.6.32-504.30.3.el6.x86_64
npm ERR! argv "/root/.nvm/versions/node/v4.0.0/bin/node" "/root/.nvm/versions/node/v4.0.0/bin/npm" "install" "zookeeper"
npm ERR! node v4.0.0
npm ERR! npm  v2.14.2
npm ERR! code ELIFECYCLE

npm ERR! zookeeper@3.4.6 install: `node-gyp rebuild`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the zookeeper@3.4.6 install script 'node-gyp rebuild'.
npm ERR! This is most likely a problem with the zookeeper package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     node-gyp rebuild
npm ERR! You can get their info via:
npm ERR!     npm owner ls zookeeper
npm ERR! There is likely additional logging output above.

npm ERR! Please include the following file with any support request:
npm ERR!     /data/dev/zk/npm-debug.log

**Comment by mdlavin** [comment]
I confirmed that it fails to install even with my fixes for node 0.12 in #108 .  I'll fix 4.0 support after I get 0.12 support merged in.

**Comment by mdlavin** [comment]
@hussion I've created a branch that should work on node 4.x.  I've run the test suite against it without trouble.  I'll run that branch in our deployment for a couple of days and then merge it into the mainline and push a new version out.  If you want to test the support (I'd really appreciate it), you can reference the branch like this:

```
"zookeeper": "git://github.com/mdlavin/node-zookeeper#node-4.0-adoption-prepublish"
```
